v2/client/client_test.go

```diff
@@ -78,6 +78,182 @@ func simpleStructuredClient(target string) client.Client {
 	return c
 }
 
+func TestClientRequest(t *testing.T) {
+	now := time.Now()
+
+	binData := []byte(`{"msg":"hello","sq":42}`)
+	structureData := []byte(
+		fmt.Sprintf(`{"data":{"msg":"hello","sq":42},"datacontenttype":"application/json","id":"AABBCCDDEE","source":"/unit/test/client","specversion":"0.3","time":%q,"type":"unit.test.client"}`,
+			now.UTC().Format(time.RFC3339Nano)))
+
+	binLen := int64(len(binData))
+	structureLen := int64(len(structureData))
+
+	testCases := map[string]struct {
+		c       func(target string) client.Client
+		event   event.Event
+		resp    *http.Response
+		want    *requestValidation
+		wantRes string
+	}{
+		"binary simple v0.3": {
```

Comment:
Why you added such tests? These are already covered by tests in integration tests. Can you move the test you made specifically for the error response somewhere in integration tests and remove the tabular test here?

Reply:
Could you point out where is the integ tests?

Reply:
in `./v2/test`

Reply:
Yeah these look  a lot like the tap tests we have

Reply:
maybe this change means there is a hole in those tests?

Reply:
It could be, maybe we need to cover more the status codes like we did recently with the receiver

Reply:
FWIW, this is something pretty easy to catch with unit test here.

Reply:
@slinkydeveloper I believe there is bug in the loopback test lib. It doesn't verify `wantResult` correctly. I don't get to further investigate this issue.